pandas/core/computation/ops.py

```diff
@@ -446,6 +447,7 @@ def evaluate(self, env, engine, parser, term_type, eval_in_python):
     def convert_values(self):
         """Convert datetimes to a comparable value in an expression.
         """
+        assert self.encoding is None, self.encoding
```

Comment:
why is this needed? The method has code for self.encoding is not None, is this unreachable?

Reply:
I think this is left over from when i tried removing encoding entirely.

Reply:
on my branch, i've removed some of the *args and **kwargs in the modules touched here and just added encoding directly as a argument in Op.__init__.